ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -698,11 +700,16 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 			if err := r.Delete(ctx, &extraHeadPodToDelete); err != nil {
 				return err
 			}
+			rayClusterExpectation.ExpectDeleteHeadPod(key, extraHeadPodToDelete.Namespace, extraHeadPodToDelete.Name)
 		}
 	}
 
 	// Reconcile worker pods now
 	for _, worker := range instance.Spec.WorkerGroupSpecs {
+		if !rayClusterExpectation.IsGroupSatisfied(key, worker.GroupName) {
+			logger.Info("reconcilePods", "RayCluster", key, "Expectation", fmt.Sprintf("NotSatisfiedHeadExpectations, reconcile group %s later", worker.GroupName))
+			//		continue
```

Comment:
I guess this should be uncomment?
```suggestion
					continue
```